tests/integration_tests/build/test_licenses.py

```diff
@@ -0,0 +1,43 @@
+
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests ensuring codebase contains neccessary licenses."""
+
+import os
+
+import pytest
+
+CHROMIUM_COPYRIGHT = "Copyright 2017 The Chromium OS Authors. All rights reserved."
+CHROMIUM_LICENSE = "Use of this source code is governed by a BSD-style license that can be"
+AMAZON_COPYRIGHT = "Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved."
+AMAZON_LICENSE = "SPDX-License-Identifier: Apache-2.0"
+EXCLUDED_DIRECTORIES = ((os.path.join(os.getcwd(), 'build')))
+
+# Python and Rust files should have the licenses on the first 2 lines.
+# Shell files should have the license on lines 3-4 to account for the shebang
+def validate_license(filename):
+    """Validate licenses in all .rs, .py. and .sh file"""
+    if filename.startswith(EXCLUDED_DIRECTORIES):
+        return True
+    if filename.endswith(('.rs', '.py', '.sh')):
+        with open(filename) as file:
+            if filename.endswith('.sh'):
+                # Rather than read whole file into memory, move iterator to third line
+                file.readline()
+                file.readline()
+            copy = file.readline()
+            local_license = file.readline()
+            return (CHROMIUM_COPYRIGHT in copy
+                    and CHROMIUM_LICENSE in local_license) or (
+                        AMAZON_COPYRIGHT in copy and AMAZON_LICENSE in local_license)
```

Comment:
Hmm, wondering if this can laid out in a more readable fashion. E.g., you could compute `has_amazon_copyright` and `has_chromium_copyright` above and then just check those in a neat return.